include/GafferScene/SetPath.h

```diff
@@ -0,0 +1,113 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2022, Cinesite VFX Ltd. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef GAFFERSCENE_SETPATH_H
+#define GAFFERSCENE_SETPATH_H
+
+#include "GafferScene/Export.h"
+#include "GafferScene/TypeIds.h"
+
+#include "Gaffer/Path.h"
+
+#include "IECore/PathMatcher.h"
+
+namespace Gaffer
+{
+
+IE_CORE_FORWARDDECLARE( Context )
+IE_CORE_FORWARDDECLARE( Plug )
+IE_CORE_FORWARDDECLARE( Node )
+
+} // namespace Gaffer
+
+namespace GafferScene
+{
+
+IE_CORE_FORWARDDECLARE( ScenePlug )
+
+class GAFFERSCENE_API SetPath : public Gaffer::Path
```

Comment:
Until such a moment as we find a general use-case for SetPath to justify its wider existence, I've moved it over to GafferSceneUI as part of SetEditorBinding.

Two fixups here, one to revert the initial addition of SetPath to GafferScene 3ee6763, and another to add it to GafferSceneUI d84b2a3. Once we're all good with this reshuffle, I'll drop the `GafferScene : Add SetPath` related commits as part of the squash.

Do you think this goes far enough or should we also have Python access to SetPath housed within `GafferSceneUI.Private`?

Reply:
> Do you think this goes far enough or should we also have Python access to SetPath housed within GafferSceneUI.Private?

I don't think it goes far enough, because it's currently accessible as `GafferSceneUI.SetPath`, which is public. The precedent from LightEditor and the other bits of SetEditor would be for it to be bound as `GafferSceneUI._SetEditorSetPath`, which is reasonably private. I think that'd do for now? Alternatively, we could make `GafferSceneUI._SetEditor.SetPath` and move the other set editor C++ components into that new namespace.